subiquity/server/controllers/integrity.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import logging
+
+from subiquitycore.async_helpers import schedule_task
+from subiquitycore.utils import astart_command
+from subiquity.common.apidef import API
+from subiquity.common.types import CasperMd5Results
+from subiquity.server.controller import SubiquityController
+
+
+log = logging.getLogger('subiquity.server.controllers.integrity')
+
+mock_pass = {'checksum_missmatch': [], 'result': 'pass'}
+mock_skip = {'checksum_missmatch': [], 'result': 'skip'}
+mock_fail = {'checksum_missmatch': ['./casper/initrd'], 'result': 'fail'}
+
+
+class IntegrityController(SubiquityController):
+
+    endpoint = API.integrity
+
+    model_name = 'integrity'
+    result_filepath = '/run/casper-md5check.json'
+
+    @property
+    def result(self):
+        return CasperMd5Results(
+                self.model.md5check_results.get('result', 'unknown'))
+
+    async def GET(self) -> CasperMd5Results:
```

Comment:
I guess at some point we may want a version of this that blocks until the check is done? But still a good start.

Reply:
An optional block, perhaps.  If a slow verifying system experiences a crash, I wouldn't want to block the dialog waiting on the result.  Or is that desired?

Reply:
no i think how you have things currently is right for this use case.